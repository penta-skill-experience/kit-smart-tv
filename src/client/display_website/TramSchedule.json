{
  "REFRESH_RATE": 600000,
  "API_KEY": "b5f142241278dc5e901f7f14f5f618a0",
  "URL": "https://live.kvv.de/webapp/departures/bystop/DTO?maxInfos=10&key=377d840e54b59adbe53608ba1aad70e8",
  "CORS_ANYWHERE": "https://cors-anywhere.herokuapp.com/",
  "ITEM_COUNT": 4
}

