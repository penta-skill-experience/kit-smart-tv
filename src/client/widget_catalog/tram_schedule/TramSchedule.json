{
  "REFRESH_RATE": 20000,
  "API_KEY": "377d840e54b59adbe53608ba1aad70e8",
  "URL_BEFORE_STOP": "https://live.kvv.de/webapp/departures/bystop/",
  "URL_AFTER_STOP": "?maxInfos=10&key=",
  "STOP": "DTO",
  "URL_STOP_SEARCH_BEFORE_STOP": "https://live.kvv.de/webapp/stops/byname/",
  "URL_STOP_SEARCH_AFTER_STOP": "?key=",
  "CORS_ANYWHERE": "https://cors-anywhere.herokuapp.com/",
  "DEFAULT_ITEM_COUNT": 4
}
