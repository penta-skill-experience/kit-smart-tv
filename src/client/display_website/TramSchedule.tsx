import * as React from "react";
import * as TramScheduleConfig from "./TramSchedule.json";
import axios from "axios"

export class TramSchedule extends React.Component<any, any> {

    constructor(props) {
        super(props);
        this.state = {
            temp: ""
        };
    }
    async getSchedule() {
        await axios.get("https://cors-anywhere.herokuapp.com/" + TramScheduleConfig.URL, {
            headers: {
                'origin': 'www.example.com'
            }
        })
            .then(resp => {
                console.log(resp.data);
                this.setState({
                    temp: resp.data
                });
            })
            .catch(function (error) {
                console.log(error);
            });
    }
    componentDidMount() {
        this.getSchedule();
        setInterval(() => this.getSchedule(), TramScheduleConfig.REFRESH_RATE);
    }

    render() {
        return <div className="grid grid-col-3 grid-flow-col gap-3 box-border">
        </div>;
    }
}