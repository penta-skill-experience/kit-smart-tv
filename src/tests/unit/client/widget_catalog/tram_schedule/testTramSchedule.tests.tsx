/**
 * @jest-environment jsdom
 */
import fetchMock from "jest-fetch-mock";
import React from "react";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {configure, shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {
    TramScheduleDisplayComponent
} from "../../../../../client/widget_catalog/tram_schedule/TramScheduleDisplayComponent";
import * as TramScheduleConfig from "../../../../../client/widget_catalog/tram_schedule/TramSchedule.json";
import {
    TramScheduleConfigComponent
} from "../../../../../client/widget_catalog/tram_schedule/TramScheduleConfigComponent";

fetchMock.enableMocks();

global.Headers = class {
    constructor() {};
    append(var1 : string, var2 : string) {};
} as jest.Mock;
configure({adapter: new Adapter()});

afterAll(() => {
    jest.restoreAllMocks();
});

describe("Tram Schedule Snapshots", () => {

    beforeEach(() => {
        fetchMock.mockResponse(
            JSON.stringify(
                {"timestamp":"2022-03-10 17:32:48","stopName":"Hauptfriedhof","departures":[{"route":"S2","destination":"Blankenloch","direction":"2","time":"3 min","lowfloor":false,"realtime":true,"traction":0},{"route":"3","destination":"Rintheim","direction":"2","time":"4 min","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Waldstadt","direction":"2","time":"5 min","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Oberreut","direction":"1","time":"6 min","lowfloor":false,"realtime":true,"traction":0},{"route":"3","destination":"Daxlanden über Hbf","direction":"1","time":"9 min","lowfloor":false,"realtime":true,"traction":0},{"route":"S2","destination":"Spöck","direction":"2","time":"17:45","lowfloor":false,"realtime":true,"traction":0},{"route":"3","destination":"Rintheim","direction":"2","time":"17:46","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Waldstadt","direction":"2","time":"17:48","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Oberreut","direction":"1","time":"17:49","lowfloor":false,"realtime":true,"traction":0},{"route":"S2","destination":"Rheinstetten","direction":"1","time":"17:50","lowfloor":false,"realtime":true,"traction":0}]}
            )
        );
    });

    test("Tram schedule testing with content", async () => {

        const configuration = {
            stop: TramScheduleConfig.DEFAULT_STOP_ID,
            count: TramScheduleConfig.DEFAULT_ITEM_COUNT,
        };
        const wrapper = shallow(<TramScheduleDisplayComponent error={(msg => {})} config={configuration} specialBoldFontColor={"ForestGreen"} specialSubtleFontColor={"DarkOrange"}/>);
        await new Promise(process.nextTick);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

    test("Tram schedule testing without content", async () => {
        const configuration = {
            stop: TramScheduleConfig.DEFAULT_STOP_ID,
            count: TramScheduleConfig.DEFAULT_ITEM_COUNT,
        };
        const wrapper = shallow(<TramScheduleDisplayComponent error={(msg => {})} config={configuration} specialBoldFontColor={"ForestGreen"} specialSubtleFontColor={"DarkOrange"}/>);
        await new Promise(process.nextTick);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});

describe("", () => {

    beforeEach(() => {
        fetchMock.mockResponse(
            JSON.stringify(
                {"timestamp":"2022-03-10 17:32:48","stopName":"Hauptfriedhof","departures":[{"route":"S2","destination":"Blankenloch","direction":"2","time":"3 min","lowfloor":false,"realtime":true,"traction":0},{"route":"3","destination":"Rintheim","direction":"2","time":"4 min","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Waldstadt","direction":"2","time":"5 min","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Oberreut","direction":"1","time":"6 min","lowfloor":false,"realtime":true,"traction":0},{"route":"3","destination":"Daxlanden über Hbf","direction":"1","time":"9 min","lowfloor":false,"realtime":true,"traction":0},{"route":"S2","destination":"Spöck","direction":"2","time":"17:45","lowfloor":false,"realtime":true,"traction":0},{"route":"3","destination":"Rintheim","direction":"2","time":"17:46","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Waldstadt","direction":"2","time":"17:48","lowfloor":false,"realtime":true,"traction":0},{"route":"4","destination":"Oberreut","direction":"1","time":"17:49","lowfloor":false,"realtime":true,"traction":0},{"route":"S2","destination":"Rheinstetten","direction":"1","time":"17:50","lowfloor":false,"realtime":true,"traction":0}]}
            )
        ).mockResponseOnce(
            JSON.stringify(
                {"stops":[{"id":"HFR","name":"Hauptfriedhof","lat":49.013053,"lon":8.431125}]}
            )
        );
    });

    test("Tram schedule config without content", async () => {
        const configuration = {
            stop: TramScheduleConfig.DEFAULT_STOP_ID,
            count: TramScheduleConfig.DEFAULT_ITEM_COUNT,
        };
        const wrapper = shallow(<TramScheduleConfigComponent config={configuration}/>);
        await new Promise(process.nextTick);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
