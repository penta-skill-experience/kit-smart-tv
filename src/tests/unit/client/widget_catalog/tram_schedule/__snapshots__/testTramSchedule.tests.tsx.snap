// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Tram schedule config without content 1`] = `
<div>
  <ForwardRef(DialogTitle)>
    Tram Schedule Settings
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "column",
          "minHeight": "300px",
          "paddingTop": "10px",
        }
      }
    >
      <ForwardRef(Autocomplete)
        autoComplete={true}
        filterSelectedOptions={true}
        inputValue=""
        onChange={[Function]}
        onInputChange={[Function]}
        options={Array []}
        renderInput={[Function]}
        sx={
          Object {
            "width": 300,
          }
        }
        value=""
      />
      <div
        style={
          Object {
            "height": 50,
          }
        }
      />
      <span>
        Number of Entries to Display:
      </span>
      <ForwardRef(Select)
        onChange={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
        value={4}
      >
        <ForwardRef(MenuItem)
          value={1}
        >
          1
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          value={2}
        >
          2
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          value={3}
        >
          3
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          value={4}
        >
          4
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          value={5}
        >
          5
        </ForwardRef(MenuItem)>
      </ForwardRef(Select)>
    </div>
  </ForwardRef(DialogContent)>
</div>
`;

exports[`Tram Schedule Snapshots Tram schedule testing with content 1`] = `
<div
  className="grid grid-flow-row sm:g-0.5 xl:gap-1.5 2xl:gap-2 box-border"
>
  <div
    className="font-light leading-none sm:text-sm sm:text-center lg:text-base xl:text-xl 2xl:text-2xl
             4xl:text-3xl 8xl:text-5xl sm:pb-1 lg:pb-2 xl:pb-3 4xl:pb-4"
  >
      
    Hauptfriedhof
  </div>
  <div
    className="font-light leading-normal sm:text-sm sm:text-left lg:text-base xl:text-xl 2xl:text-2xl 4xl:text-3xl 8xl:text-5xl"
  >
    S2
     
    Blankenloch
    :    
    3 min
  </div>
  <div
    className="font-light leading-normal sm:text-xs lg:text-base xl:text-base 2xl:text-xl 4xl:text-2xl sm:text-left 8xl:text-4xl"
    key="0"
  >
    3
     
    Rintheim
    :    
    4 min
  </div>
  <div
    className="font-light leading-normal sm:text-xs lg:text-base xl:text-base 2xl:text-xl 4xl:text-2xl sm:text-left 8xl:text-4xl"
    key="1"
  >
    4
     
    Waldstadt
    :    
    5 min
  </div>
  <div
    className="font-light leading-normal sm:text-xs lg:text-base xl:text-base 2xl:text-xl 4xl:text-2xl sm:text-left 8xl:text-4xl"
    key="2"
  >
    4
     
    Oberreut
    :    
    6 min
  </div>
</div>
`;

exports[`Tram Schedule Snapshots Tram schedule testing without content 1`] = `
<div
  className="grid grid-flow-row sm:g-0.5 xl:gap-1.5 2xl:gap-2 box-border"
>
  <div
    className="font-light leading-none sm:text-sm sm:text-center lg:text-base xl:text-xl 2xl:text-2xl
             4xl:text-3xl 8xl:text-5xl sm:pb-1 lg:pb-2 xl:pb-3 4xl:pb-4"
  >
      
    Hauptfriedhof
  </div>
  <div
    className="font-light leading-normal sm:text-sm sm:text-left lg:text-base xl:text-xl 2xl:text-2xl 4xl:text-3xl 8xl:text-5xl"
  >
    S2
     
    Blankenloch
    :    
    3 min
  </div>
  <div
    className="font-light leading-normal sm:text-xs lg:text-base xl:text-base 2xl:text-xl 4xl:text-2xl sm:text-left 8xl:text-4xl"
    key="0"
  >
    3
     
    Rintheim
    :    
    4 min
  </div>
  <div
    className="font-light leading-normal sm:text-xs lg:text-base xl:text-base 2xl:text-xl 4xl:text-2xl sm:text-left 8xl:text-4xl"
    key="1"
  >
    4
     
    Waldstadt
    :    
    5 min
  </div>
  <div
    className="font-light leading-normal sm:text-xs lg:text-base xl:text-base 2xl:text-xl 4xl:text-2xl sm:text-left 8xl:text-4xl"
    key="2"
  >
    4
     
    Oberreut
    :    
    6 min
  </div>
</div>
`;
